src/Common/src/Interop/Unix/libssl/SecuritySafeHandles.cs

```diff
@@ -249,11 +284,10 @@ public SafeDeleteContext(SafeFreeCredentials credential, bool isServer, bool rem
             }
             catch(Exception ex)
             {
-                if (gotCredRef)
-                {
-                    _credential.DangerousRelease();
-                }
                 Debug.Write("Exception Caught. - " + ex);
+                Debug.Assert((null != credential), "Null credential in SafeDeleteContext");
+                credential.DangerousRelease();
```

Comment:
See my previous response.  This needs to be fixed, as it's potentially double-releasing.